app/presenters/summary/html_sections/base_children_details.rb

```diff
@@ -32,24 +32,61 @@ def personal_details(child)
 
       def relationships(child)
         [
-          MultiAnswer.new(:child_applicants_relationship,  relation_to_child(child, c100.applicants), change_path: edit_relation_path(child, c100.applicants)),
-          MultiAnswer.new(:child_respondents_relationship, relation_to_child(child, c100.respondents), change_path: edit_relation_path(child, c100.applicants)),
+          MultiAnswer.new(:child_applicants_relationship,
+                          relation_to_child(child, c100.applicants),
+                          change_path: edit_relation_path(child,
+                                                          c100.applicants)),
+          MultiAnswer.new(:child_respondents_relationship,
+                          relation_to_child(child, c100.respondents),
+                          change_path: edit_relation_path(child,
+                                                          c100.respondents)),
         ]
       end
 
       private
 
       def edit_relation_path(child, people)
         format("/steps/applicant/relationship/%<id>s/child/%<child_id>s",
-               id: relationship(child, people).pluck(:id),
+               id: relationship(child, people).pluck(:person_id).first,
                child_id: child.id)
       end
 
+      def child_names_method(child)
```

Comment:
Not saying to do do it, but maybe it would better to have all these route methods in subclasses, (so we would have a children subclass and an other_children subclass) so the subclasses are in charge of knowing the routes, and then in the superclass just call a generic method, lie 'edit_names_path', 'edit_personal_details_path' and so on...

These if-else are going to be messy for applicants, respondents, other parties (and solicitors, when we introduce them). Better to encapsulate anything that is particular to each kind of 'Person' in their subclasses, and use a template-pattern.

Reply:
you're right, I've refactored it and it's much cleaner now :+1: